File: Ejercicio2.py


def verificar_contrasena():
    contrasena_almacenada = "LOL"
    contrasena_usuario = input("Por favor, ingresa la contraseña: ")

    if contrasena_almacenada.lower() == contrasena_usuario.lower():
        print("Las contraseñas coinciden.")
    else:
        print("Las contraseñas NO coinciden.")

if __name__ == "__main__":
    verificar_contrasena()

File: Ejercicio2.hs
module Ejercicio2 where

import Data.Char (toLower)
import System.IO


verificarContrasena :: String -> String -> Bool
verificarContrasena almacenada entrada = (map toLower almacenada) == (map toLower entrada)

main :: IO ()
main = do
    hSetBuffering stdout NoBuffering
    putStr "Ingrese la contraseña almacenada: "
    passAlmacenada <- getLine
    putStr "Ingrese la contraseña a verificar: "
    passEntrada <- getLine
    let resultado = verificarContrasena passAlmacenada passEntrada
    putStrLn $ "Las contraseñas coinciden: " ++ show resultado


